processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/GroupByQueryEngine.java

```diff
@@ -219,16 +220,16 @@ private static boolean hasNoImplicitUnnestDimensions(
       }
 
       if (dimension.getOutputType().isArray()) {
-        return false;
+        continue;
       }
 
       // Now check column capabilities, which must be present and explicitly not multi-valued and not arrays
       final ColumnCapabilities capabilities = inspector.getColumnCapabilities(dimension.getDimension());
-      if (capabilities != null && capabilities.hasMultipleValues().isFalse() && !capabilities.isArray()) {
+      if (capabilities != null && capabilities.hasMultipleValues().isMaybeTrue() && !capabilities.isArray()) {
         return false;
       }
     }
-    return true;
+    return !hasImplicitUnnest;
```

Comment:
`hasImplicitUnnest` is always false?

Reply:
oops, was messing around with stuff and didn't mean to leave this here